tests/components/suez_water/test_sensor.py

```diff
@@ -0,0 +1,92 @@
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers import entity_registry as er
+
+from . import setup_integration
+
+from tests.common import MockConfigEntry, async_fire_time_changed, snapshot_platform
+
+
+async def test_suez_sensors_valid_state(
+    hass: HomeAssistant,
+    snapshot: SnapshotAssertion,
+    suez_client: MagicMock,
+    mock_config_entry: MockConfigEntry,
+    entity_registry: er.EntityRegistry,
+) -> None:
+    """Test that suez_water sensor is loaded and in a valid state."""
+    with patch("homeassistant.components.suez_water.PLATFORMS", [Platform.SENSOR]):
+        await setup_integration(hass, mock_config_entry)
+
+    assert mock_config_entry.state is ConfigEntryState.LOADED
+    await snapshot_platform(hass, entity_registry, snapshot, mock_config_entry.entry_id)
+
+
+async def test_suez_sensors_failed_update(
+    hass: HomeAssistant,
+    suez_client,
+    mock_config_entry: MockConfigEntry,
+    freezer: FrozenDateTimeFactory,
+) -> None:
+    """Test that suez_water sensor reflect failure when api fails."""
+
+    with patch("homeassistant.components.suez_water.PLATFORMS", [Platform.SENSOR]):
+        await setup_integration(hass, mock_config_entry)
+
+        assert mock_config_entry.state is ConfigEntryState.LOADED
+
+        entity_ids = await hass.async_add_executor_job(hass.states.entity_ids)
+        assert len(entity_ids) == 1
+
+        state = hass.states.get(entity_ids[0])
+        assert entity_ids[0]
+        assert state.state != STATE_UNAVAILABLE
+
+        suez_client.update.side_effect = PySuezError("Should fail to update")
+
+        freezer.tick(timedelta(hours=1) + DATA_REFRESH_INTERVAL)
+        async_fire_time_changed(hass)
+        await hass.async_block_till_done(True)
+
+        state = hass.states.get(entity_ids[0])
+        assert state
+        assert state.state == STATE_UNAVAILABLE
+
+
+async def test_suez_sensors_invalid_credentials(
+    hass: HomeAssistant,
+    suez_client,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test that suez_water sensor can't be loaded with invalid credentials."""
+
+    with patch("homeassistant.components.suez_water.PLATFORMS", [Platform.SENSOR]):
+        suez_client.check_credentials.return_value = False
+        await setup_integration(hass, mock_config_entry)
+
+        assert mock_config_entry.state is ConfigEntryState.SETUP_ERROR
+
+
+async def test_suez_sensors_setup_api_error(
+    hass: HomeAssistant,
+    suez_client,
+    mock_config_entry: MockConfigEntry,
+) -> None:
+    """Test that suez_water sensor needs to retry loading if api failed to connect."""
+
+    with patch("homeassistant.components.suez_water.PLATFORMS", [Platform.SENSOR]):
+        suez_client.check_credentials.side_effect = PySuezError("Test failure")
+        await setup_integration(hass, mock_config_entry)
+
+        assert mock_config_entry.state is ConfigEntryState.SETUP_RETRY
```

Comment:
These 2 are for `test_init.py`, since they don't test sensors, they test if the integration is able to set up. So the name and docstrings should also be updated